parts/k8s/addons/kubernetesmasteraddons-kube-dns-deployment.yaml

```diff
@@ -47,6 +47,8 @@ spec:
     metadata:
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        prometheus.io/scrape: "true"
```

Comment:
Should these annotations be on the service rather than the deployment?

Reply:
I believe the recommended approach is to annotate the service (though I cannot find a black and white statement from the team about it now). The pods of the service will still get scraped if using a `role` of `endpoints` in the `kubernetes_sd_configs`.
The exposed metrics will be the same either way, but there are a bunch of extra labels available this way.

Also, do note that this annotation is not really standardized, even if it is common in examples. We do not use it at all, and rather use `ServiceMonitor`s from the prometheus-operator. But perhaps it is a reasonable default to ship anyway? Is it set on other kube* things with metrics in acs-engine?